types/eslint-plugin-prettier/index.d.ts

```diff
@@ -0,0 +1,36 @@
+// Type definitions for eslint-plugin-prettier 2.2
+// Project: https://github.com/prettier/eslint-plugin-prettier
+// Definitions by: Ika <https://github.com/ikatyang>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+/**
+ * Converts invisible characters to a commonly recognizable visible form.
+ * @param str The string with invisibles to convert.
+ */
+export function showInvisibles(str: string): string;
+
+/**
+ * Generate results for differences between source code and formatted version.
+ * @param source The original source.
+ * @param formatted The formatted source.
+ */
+export function generateDifferences(
+    source: string,
+    formatted: string,
+): Difference[];
+
+export interface Difference {
+    operation: DifferenceOperation;
```

Comment:
I would define this as `"insert" | "delete" | "replace"`. const enums have limitations when used with `--isolatedModules` and i would not recommend them in this context.